### PR TITLE
update karpenter docs to working example.  remove taints, podIdentity…

### PR DESCRIPTION
**Pull request by zjaco13**
Update karpenter docs to working example.  Remove taints, podIdentity, and budgets.  Switch spot to on-demand

*Issue #, if available:*
#1038

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by elamaran11 — APPROVED** [review]
LGTM

File: docs/addons/karpenter.md [hunk 1]
```diff
@@ -38,37 +40,32 @@ const karpenterAddOn = new blueprints.addons.KarpenterAddOn({
...
-      taints: [{
-          key: "workload",
-          value: "test",
-          effect: "NoSchedule",
-      }],
       requirements: [
-          { key: 'node.kubernetes.io/instance-type', operator: 'In', values: ['m5.2xlarge'] },
-          { key: 'topology.kubernetes.io/zone', operator: 'In', values: ['us-west-2a','us-west-2b', 'us-west-2c']},
+          { key: 'node.kubernetes.io/instance-type', operator: 'In', values: ['m5.large'] },
+          { key: 'topology.kubernetes.io/zone', operator: 'In', values: [`${region}a`, `${region}b`, `${region}c`] },
```
Comment (shapirov103) [inline docs/addons/karpenter.md:45]: Should these be having a dash before the zone: `${region}-a` as opposed to `${region}a`?

File: docs/addons/karpenter.md [hunk 2]
```diff
@@ -38,37 +40,32 @@ const karpenterAddOn = new blueprints.addons.KarpenterAddOn({
...
           { key: 'kubernetes.io/arch', operator: 'In', values: ['amd64','arm64']},
-          { key: 'karpenter.sh/capacity-type', operator: 'In', values: ['spot']},
+          { key: 'karpenter.sh/capacity-type', operator: 'In', values: ['on-demand']},
```
Comment (shapirov103) [inline docs/addons/karpenter.md:47]: good one, let's add a comment here that they can use spot if need to. Very handy for cost saving for some folks.

File: docs/addons/karpenter.md [hunk 3]
```diff
@@ -38,37 +40,32 @@ const karpenterAddOn = new blueprints.addons.KarpenterAddOn({
       annotations: {
           "eks-blueprints/owner": "young"
```
Comment (shapirov103) [inline docs/addons/karpenter.md:41]: I know it is not authored by you, but let' replace the owner with something generic, e.g. "platform-team"

**Review by shapirov103 — COMMENTED** [review]
Awesome work, please see a few minor things.

**Review by shapirov103 — CHANGES_REQUESTED** [review]
See my comments

File: docs/addons/karpenter.md (hunk 1 above)
Comment (zjaco13) [reply docs/addons/karpenter.md:45]: No dash should have it resolve to us-west-2a, us-west-2b, and us-west-2c, which matches the example

Comment (shapirov103) [reply docs/addons/karpenter.md:45]: nm, not sure what I am seeing in the morning.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**